src/slots/factories/SlotFactory.js

```diff
@@ -15,15 +27,48 @@ class SlotFactory {
     return SlotFactory.SUPPORTED_CLASSES.includes(Class)
   }
 
-  static get TYPE_TO_CLASS() { return ({ InSlot, OutSlot }) }
+  static get DATATYPE_TO_IN_SLOT_CLASS() {
+    return {
+      [IntegerInSlot.DATA_TYPE]: IntegerInSlot,
+      [FloatInSlot.DATA_TYPE]: FloatInSlot,
+      [BooleanInSlot.DATA_TYPE]: BooleanInSlot,
+      [OneOfInSlot.DATA_TYPE]: OneOfInSlot,
+    }
+  }
+
+  static get DATATYPE_TO_OUT_SLOT_CLASS() {
+    return {
+      [OutSlot.DATA_TYPES.INTEGER]: OutSlot,
+      [OutSlot.DATA_TYPES.FLOAT]: OutSlot,
+    }
+  }
```

Comment:
we very well may want/need/decide to do the same thing with OutSlots that you are doing now with InSlots in the future and this sets that potential change up really nicely! kind programming.